Library/Homebrew/cask/info.rb

```diff
@@ -31,12 +31,12 @@ def self.get_info(cask)
       output
     end
 
-    sig { params(cask: Cask).void }
-    def self.info(cask)
+    sig { params(cask: Cask, args: Homebrew::Cmd::Info::Args).void }
+    def self.info(cask, args:)
       puts get_info(cask)
 
       require "utils/analytics"
-      ::Utils::Analytics.cask_output(cask, args: Homebrew::CLI::Args.new)
+      ::Utils::Analytics.cask_output(cask, args:)
```

Comment:
I believe this was an existing bug that only came to light because of recent changes. It seems like by creating a new instance of the CLI args here that `args.analytics?` would always be falsey later on which doesn't seem to be the desired behavior. Now when the `--analytics` flag is set it will get passed to `Utils::Analytics.cask_output` the same way it already gets passed to `Utils::Analytics.formula_output`.

Reply:
Nice work! We might want to add some guardrails around where/how `Args` can be instantiated 🤔

Reply:
IMO it's probably not necessary unless it's really simple to implement since I doubt that we'll ever run into the same problem again.